joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstCreator.scala

```diff
@@ -103,4 +115,70 @@ class AstCreator(
   protected def astForNode(json: Value): Seq[Ast] = {
     astForNode(createParserNodeInfo(json))
   }
+
+  def init(): ParserResult = {
+    val parserResult = GoAstJsonParser.readFile(Paths.get(jsonAstFilePath))
+    lineNumberMapping = positionLookupTables(parserResult)
+    declaredPackageName = parserResult.json(ParserKeys.Name)(ParserKeys.Name).str
+    fullyQualifiedPackage = goMod.getNameSpace(parserResult.fullPath, declaredPackageName)
+    fileName = parserResult.filename
+    originalFilePath = parserResult.fullPath
+    parserResult
+  }
+
+  def cacheSerializeAndStore(): Unit = {
+    tempAliasToNameSpaceMappingFilePath.map(file => {
+      Files.write(file, serialise(aliasToNameSpaceMapping))
+      aliasToNameSpaceMapping.clear()
+    })
+    lineNumberMapping = Map.empty
+  }
+
+  def loadCacheToProcess(): Unit = {
+    tempAliasToNameSpaceMappingFilePath.map(file => {
+      val deserialised = deserialise(Files.readAllBytes(file))
+      aliasToNameSpaceMapping = deserialised.asInstanceOf[mutable.Map[String, String]]
+    })
+  }
+
+  def cleanup(): Unit = {
```

Comment:
I would rather suggest using `close` and having `AstCreator` extend `AutoCloseable`, this can enforce `Using.resource(astCreator) => ` which guarantees this global mutable state is at least closed otherwise we may sit with an implicit protocol which may not be followed and result in a memory leak.